design/general-global-rate-limit-policy.md

```diff
@@ -0,0 +1,100 @@
+# General Global RateLimit Policy
+
+Status: Draft
+
+## Abstract
+Define a general rate limit policy in the Contour configuration to be used as a global rate limit policy by all HTTPProxy.
+
+## Background
+See https://github.com/projectcontour/contour/issues/5357 for context.
+
+## Goals
+- Define a general L7 global rate limit policy which is used by all HTTPProxy on the `VirtualHost` level.
+
+## Non Goals
+- General rate limit policy for Route level `rateLimitPolicy`.
+- L4 rate limiting support. 
+
+## High-Level Design
+Currently, the global `rateLimitPolicy` for `VirtualHost` is set per HTTPProxy. This option allows each service to configure its own global `rateLimitPolicy`. However, for a single tenant setup/general usecase, this means you have to edit all the HTTPProxy objects with the same ratelimit policies.
+
+A new field, `GeneralRateLimitPolicy` (optional), will be added to `RateLimitServiceConfig` which is part of `ContourConfigurationSpec`. This field will define rate limit descriptors that will be added to every `VirtualHost`.
+
+HTTPProxy has to opt-in **explicitly** to use the general rate limit policy using `rateLimitPolicy.generalRateLimitPolicyEnabled` flag and still can have its own global `rateLimitPolicy` which overrides the general one.
```

Comment:
I think it's going to be more consistent to do this the other way, i.e. to allow individual HTTPProxies to **opt out** of the general rate limit policy if it's defined. This is what we did with the recent global ExtAuth work, for example. Is there a specific reason you proposed using **opt-in** here?

Reply:
I was just thinking about not introducing a regression. That is it.

And then I was like If I am gonna add `DefaultGlobalRateLimitPolicy` I properly would like this policy to be applied directly without the need to edit all the HTTPProxy and opt them in manually.
So I don't really mind opt-in by default (in fact I would prefer this option) But yeah, the main motivation is just not introducing regression.

Reply:
+1 on `opt-out` as the default but I do see Shadi's point on breaking changes. What is the pattern here?

Reply:
In this case, since `DefaultGlobalRateLimitPolicy` will be a _new_ config field that is not populated by default, it should be backwards-compatible for folks who aren't using the feature, so I'd say having it apply to all HTTPProxies by default (i.e. opt-out behavior) if it is populated is fine and probably a better UX overall, as long as there's also a way to opt individual HTTPProxies out. cc @sunjayBhatia @tsaarni for other thoughts.

Reply:
yep +1 to that ^ 

generally we've tried to have the global configs set in config file/CRD as operator owned defaults that can be overridden for individual HTTPProxies as needed so that should be consistent

Reply:
Yeah I think the flag of opting out should be part of the Global rate limit policy for consistency